frontend/src/components/widget/WidgetRenderer.js

```diff
@@ -0,0 +1,629 @@
+
+    const listAndLookupsProps = {
+      dataId,
+      attribute,
+      entity,
+      subentity,
+      subentityId,
+      windowType,
+      readonly,
+      updated,
+      filterWidget,
+      filterId,
+      tabId,
+      rowId,
+      tabIndex,
+      viewId,
+      autoFocus,
+      onFocus: onListFocus,
+      onBlur: onBlurWithParams,
+      align: gridAlign,
+      mandatory: widgetData[0].mandatory,
+      parameterName: fields[0].parameterName,
+      validStatus: widgetData[0].validStatus,
+      onChange: onPatch,
+    };
+    const dateProps = {
+      field: widgetField,
+      key: 1,
+      handleChange,
+      handleBackdropLock,
+      inputProps: {
+        placeholder: fields[0].emptyText,
+        disabled: readonly,
+        tabIndex: tabIndex,
+      },
+      onChange: (date) => handleChange(widgetField, date),
+    };
+    const dateRangeProps = {
+      mandatory: widgetData[0].mandatory,
+      validStatus: widgetData[0].validStatus,
+      value: widgetData[0].value,
+      valueTo: widgetData[0].valueTo,
+      tabIndex,
+      onShow,
+      onHide,
+    };
+    const attributesProps = {
+      entity,
+      fields,
+      dataId,
+      widgetData: widgetData[0],
+      docType: windowType,
+      tabId,
+      rowId,
+      fieldName: widgetField,
+      handleBackdropLock,
+      patch: (option) => onPatch(widgetField, option),
+      tabIndex,
+      autoFocus,
+      readonly,
+    };
+
+    switch (widgetType) {
+      case 'Date':
+        if (range) {
+          // TODO: Watch out! The datetimerange widget as exception,
+          // is non-controlled input! For further usage, needs
+          // upgrade.
+          return (
+            <DatetimeRange
+              {...dateRangeProps}
+              onChange={(value, valueTo) =>
+                onPatch(
+                  widgetField,
+                  value ? Moment(value).format(DATE_FORMAT) : null,
+                  null,
+                  valueTo ? Moment(valueTo).format(DATE_FORMAT) : null
+                )
+              }
+              field={widgetField}
+              timeZone={timeZone}
+            />
+          );
+        } else {
+          return (
+            <div className={this.getClassNames({ icon: true })}>
+              <DatePicker
+                {...dateProps}
+                timeFormat={false}
+                dateFormat={dateFormat || true}
+                value={widgetValue || widgetData[0].value}
+                patch={(date) =>
+                  onPatch(
+                    widgetField,
+                    this.generateMomentObj(date, DATE_FORMAT),
+                    null,
+                    null,
+                    true
+                  )
+                }
+                {...{
+                  isOpenDatePicker,
+                  timeZone,
+                }}
+              />
+            </div>
+          );
+        }
+      case 'ZonedDateTime':
+        return (
```

Comment:
Would be nice to separate here to have <ZoneDateTime {...props} /> and have ZoneDateTime defined in a separate component